meerkat_frontend/src/js/technical/tabletSubmission.js

```diff
@@ -0,0 +1,320 @@
+                                        });
+                                        $.getJSON(api_root + month_Api1, function(data) {
+                                            $.each(data.clinicSubmissions, function(index, value) {
+                                                buildMonthData(value, 1, 'casesInMonthL');
+                                            });
+                                            $.getJSON(api_root + month_Api, function(data) {
+                                                $.each(data.clinicSubmissions, function(index, value) {
+                                                    buildMonthData(value, 0, 'casesInMonthC');
+                                                });
+                                                $.getJSON(api_root + locationAPI, function(data) {
+                                                    $.each(data, function(index, value) {
+                                                        getClinicCaseAndType(value);
+                                                    });
+                                                    //Append the table and the columns ...
+                                                    $('#div_tabletCompletness').append('<table  id="tbleTabletSubmission"></table>');
+                                                    $('#div_tabletCompletness table').bootstrapTable({
+                                                        columns: columnNameArray,
+                                                        data: tabletDataArray,
+                                                        pagination: true,
+                                                        pageSize: 10,
+                                                        search: true,
+                                                        classes: "table table-no-bordered table-hover table-responsive"
+                                                    });
+                                                    addPaginationListener('#clinicsTable table');
+                                                    colorClinicsRow();
+                                                });
+
+                                            });
+                                        });
+                                    });
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+    });
+}
+
+
+function buildTableStructure() {
+    //Add the first columns
+    buildTableColumn("DevicesIDs", "Devices IDs", "");
+    buildTableColumn("clinicType", "clinic type", "");
+    buildTableColumn("caseType", "case type", "");
+
+    for (i = week - 2; i <= week; i++) {
+        //Draw the clolumns depend on the week number
+        buildTableColumn("casesInWeek" + i, "cases in Week " + i, "");
+    }
+
+    for (i = week - 2; i <= week; i++) {
```

Comment:
This is the very same loop as the one from `124`. You can merge them and do two operations inside a single loop.

Reply:
so what i did here is to separate the loop so i can have the columns order like (casesInWeek 40 -casesInWeek 41 - casesInWeek 42) then (noOfRegistersInWeek 40 - ... )
so if i merge it will look like (caseInWeek 40 - noOfRegistersInWeek 40 - ...)

Reply:
This pattern of looping over three column and creating headers repeats everywhere. It could be extracted as a method:
```js
function createTripletsColumnHeaders(idPrefix, labelPrefix, enumerator) {
  for (i = enumerator - 2; i <= enumerator; i++) {
          buildTableColumn(idPrefix + i, labelPrefix + i, "");
  }
}
```
With later usage:
```js
createTripletsColumnHeaders("casesInWeek", "cases in week ", week);
createTripletsColumnHeaders("noOfRegistersInWeek", "no. of registers in week  ", week);
createTripletsColumnHeaders("casesInYear", "cases in year  ", current_Year);
```
I think it looks much more readable.